### DISCUSSION TITLE
Multiple values in custom vield

### DISCUSSION THREAD

**Discussion by joernheissler** [netbox-community/netbox#12077] [Q&A]
Hello,
I'd like to create a custom field where I can store multiple values. Is that possible? If not, is there already an issue (couldn'd find any)?

**Comment by candlerb** [comment]
What kind of values? If it's an object value, then there is multi-object. If it's a selection field, then there is multi-select.

If you want a list of arbitrary string values, then:
* you could put them in a text field as `foo,bar` or `foo|bar`
* you could put them in a JSON field as `["foo","bar"]`

A list of numbers could also go in a JSON field as `[3,42]`. And you could add a [custom validator](https://docs.netbox.dev/en/stable/customization/custom-validation/) to check it.

Anything more sophisticated would probably store internally in JSON anyway. It would then be a question of how to make the UI work with multiple values (add/remove values, validate individual values etc) which sounds like it could be quite tricky.

**Reply by joernheissler** [reply]
It would be a list of DNS aliases, i.e. `list[str]` with a custom validator.
Inside the database it could be serialized as:

`custom_field_data=`
```json
{
  "dns_aliases": [
    "host0",
    "host.with.dots",
    "host-two"
  ]
}
```

So I guess what I'm really asking for is a UI change: Instead of editing JSON directly, the UI would provide a simple list editor: One textbox per item, buttons to add/remove items, per-item validator.
